kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyConfigConfigMap.java

```diff
@@ -306,19 +326,99 @@ private static GenericKubernetesResource filterResourceFromRef(VirtualKafkaClust
+        }
+        return Optional.empty();
+    }
+
+    @NonNull
+    private static Optional<Tls> buildTargetClusterTls(Context<KafkaProxy> context,
+                                                       VirtualKafkaCluster cluster) {
+        return Optional.ofNullable(cluster.getSpec())
+                .map(VirtualKafkaClusterSpec::getTargetCluster)
+                .map(io.kroxylicious.kubernetes.api.v1alpha1.virtualkafkaclusterspec.TargetCluster::getTls)
+                .map(io.kroxylicious.kubernetes.api.v1alpha1.virtualkafkaclusterspec.targetcluster.Tls::getCertificateRef)
+                .map(certRef -> {
+                    return new Tls(new KeyPair(
+                            virtualClusterTargetTlsKeyPath(context, cluster, certRef).containerPath().toString(),
+                            virtualClusterTargetTlsCertPath(context, cluster, certRef).containerPath().toString(),
+                            null),
+                            null, null, null);
+                });
+    }
+
+    private static ContainerFileReference virtualClusterTargetTlsKeyPath(Context<KafkaProxy> context,
+                                                                         VirtualKafkaCluster cluster,
+                                                                         CertificateRef certRef) {
+        return virtualClusterTargetTlsPath(context, cluster, certRef, "tls.key");
+    }
+
+    private static ContainerFileReference virtualClusterTargetTlsCertPath(Context<KafkaProxy> context,
+                                                                          VirtualKafkaCluster cluster,
+                                                                          CertificateRef certRef) {
+        return virtualClusterTargetTlsPath(context, cluster, certRef, "tls.crt");
+    }
+
+    private static ContainerFileReference virtualClusterTargetTlsPath(Context<KafkaProxy> context,
+                                                                      VirtualKafkaCluster cluster,
+                                                                      CertificateRef certRef,
+                                                                      String file) {
+        if ("Secret".equals(certRef.getKind())
+                && certRef.getGroup().isEmpty()) {
+            // Assume it's compatible with a Secret of `type: kubernetes.io/tls`
+            String other = "vc-target-tls-" + ResourcesUtil.name(cluster);
+            Path volumeMountPath = Path.of("/opt/kroxylicious/cluster")
+                    .resolve(ResourcesUtil.name(cluster))
+                    .resolve("target")
+                    .resolve("tls-key");
+            Volume volume = new VolumeBuilder()
+                    .withName(other)
+                    .withNewSecret()
+                    .withSecretName(certRef.getName())
+                    .addNewItem("tls.key", null, "tls.key")
+                    .addNewItem("tls.crt", null, "tls.crt")
```

Comment:
Using `items` means the volume won't get created if the Secret doesn't contain the given keys (according to [this](https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/#project-secret-keys-to-specific-file-paths)), which means the pod can't be created and the Deployment wouldn't be ready.